### DISCUSSION TITLE
What does the 3.4.0 Libre app use to detect root apps?

### DISCUSSION THREAD

**Discussion by yoshimo** [maheini/FreeStyle-Libre-3-patch#13] [Q&A] [answered]
I was wondering how the official app crashes only on rooted systems.
I use the Magisk 25 feature to hide root from applications a and the rootbeer test app says "device not rooted".
Libre3 crashes , with or without LSPosed and FreeThree Module.
Did anyone do further tests on this matter?

**Comment by maheini** [comment, marked as answer]
I guess, libre 3 is not only checking for root, it's library is also checking various other things. One of them is checking Magisk.

You have a step-for-step tutorial on how to hide root properly here: [link](https://www.droidwin.com/how-to-hide-root-from-apps-via-magisk-denylist/)

I'd really apprechiate if you could report if it did work - thanks in advance. I do have big hopes into step 4, hiding magisk app - so don't miss it 😄.

**Reply by yoshimo** [reply]
Magisk is renaming itself to a random package name and i was using the deny list.
So there is something else on top.

**Reply by maheini** [reply]
Ok, very convenient from Magisk. I found a new solution, which at least successfully prevented the crash when selecting the carbs.

Here you go: [Reddit link](https://www.reddit.com/r/Freestylelibre/comments/s22vlr/comment/hw2p4th/?utm_source=share&utm_medium=web2x&context=3)


Some background: With Magisk 24 the feature called "Magisk Hide" has been removed. It was used to hide the fact that phone was rooted from certain apps. Instead and as replacement, there are two new mechanisms for that. First, the "Zygisk" mode provides the means for Magisk modules to be directly injected into any app process when it launches. That yields much more possibilities for modules as they can easily access everything that happens inside the app. Second, the new Shamiko module uses the DenyList to hide Magisk and root from any app on that List. It does so by using the new Zygisk functionality.

**Reply by yoshimo** [reply]
Mhm, what do we do with android 8? LSposed is not compatible unfortunately.

**Reply by maheini** [reply]
As far as I know, there's currently no option without LSposed - sadly 😞 

I hope the informations where useful - if you get any new approach, feel free to submit them 😄
